pandas/core/arrays/string_arrow.py

```diff
@@ -0,0 +1,486 @@
+from __future__ import annotations
+
+from collections import abc
+from distutils.version import LooseVersion
+from typing import Any, Optional, Sequence, Type, Union
+
+import numpy as np
+
+from pandas._libs import lib, missing as libmissing
+from pandas._typing import ArrayLike
+
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.dtypes import register_extension_dtype
+
+import pandas as pd
+from pandas.api.types import (
+    is_array_like,
+    is_bool_dtype,
+    is_integer,
+    is_integer_dtype,
+    is_scalar,
+)
+from pandas.core.arrays.base import ExtensionArray
+from pandas.core.indexers import check_array_indexer
+
+try:
+    import pyarrow as pa
+except ImportError:
+    pa = None
+else:
+    try:
+        import pyarrow.compute as pc
+    except ImportError:
+        pass
+
+
+def _as_pandas_scalar(arrow_scalar: pa.Scalar) -> Optional[str]:
+    scalar = arrow_scalar.as_py()
+    if scalar is None:
+        return libmissing.NA
```

Comment:
@xhochy why is this a top level function? I think we want to return self.na_value from ArrowStringArray here.

Reply:
Yes, we can do. This is probably an artifact as I started here using `fletcher` code that predates `pd.NA`.

Reply:
makes sense. thanks.